### DISCUSSION TITLE
Support for 64-bit integers

### DISCUSSION THREAD

**Discussion by adamthehutt** [livewire/livewire#1857] [4. General]
Livewire is pretty amazing, but a significant limitation is that it really struggles with very large 64-bit integers (such as model identifiers with the BIGINT type). When those integers exceed the javascript maximum of 2^53 - 1, they are automatically rounded off when cast to JSON, which causes lots of things to fail (inconsistent state between browser and component data, invalid URLs, etc.)

Currently I'm working around this by casting everything manually to strings. But this results in a LOT of boilerplate code and really undermines Livewire's awesomeness. It's impossible to use Eloquent models or model collections as public properties, for example. When calling methods like $this->emit() from a component, I need to always remember to cast any large integers to strings, etc.

I don't know how difficult this would be, since I'm not very familiar with Livewire internals. But would it be possible to intercept the basic hydration and encoding logic to automatically cast integers that exceed javascript's maximum to strings when encoding to JSON and back to integers when hydrating PHP?

Since I can't offer a pull request, I would be open to sponsoring this feature/improvement if someone wanted to take it on.

**Comment by austenc** [comment]
Interesting @adamthehutt, 

Do you think you might be able to make a [Laravel Playground](https://laravelplayground.com/) or example repo depicting the problem? Seems like this might be a nice candidate for a PR.

**Reply by nuernbergerA** [reply]
we could easily convert `bigint` to `string` when dehydrating

**Reply by adamthehutt** [reply]
Thanks for the replies. 

Not sure about Laravel Playground, but it should be simple to replicate. Create a model and assign it an ID like 200000000000000000. Then use it as a public property in a Livewire component and send some messages back and forth. 

I was reading about dehydrating in the docs, but it wasn't totally clear (to me, anyway) how that worked. Is "dehydrating" basically what Livewire calls it when it encodes data to json for transmission to the browser?

**Reply by adamthehutt** [reply]
@austenc @nuernbergerA Any thoughts on this? If you can show me where I should be looking I can try to put together a pull request.

**Comment by austenc** [comment]
I just submitted a PR for this. Great suggestion @nuernbergerA -- hopefully my PR is what you had in mind. 

@adamthehutt if you use the version of livewire in this branch from my fork, does it solve your issue?
https://github.com/austenc/livewire/tree/support-big-integers

**Reply by adamthehutt** [reply]
@austenc Thanks so much for taking this on!

I tried to do a very simple test, but I'm still getting a CorruptComponentPayloadException every time there's a data round-trip. It seems to work fine when the public property itself is the big integer, but it breaks when it's a property on an eloquent model. 

I'll comment on the PR with a bit more detail.

**Comment by bimsonz** [comment]
Recently ran into this issue when using discord role IDs as an ID for a select/ choices style component via MaryUI. Im getting rounding to the 16th on a 19 digit int. Did this gain any traction?
